dags/clean/clean.py

```diff
@@ -0,0 +1,47 @@
+from airflow.operators.bash_operator import BashOperator
+from airflow import DAG
+import pendulum
+from airflow.decorators import dag
+from airflow.operators.bash import BashOperator
+import os
+
+AIRFLOW_HOME = os.getenv("AIRFLOW_HOME")
+LOG_DIR = "logs"
+logs_dir = f'{AIRFLOW_HOME}/{LOG_DIR}'
+
+default_args = {
+    'owner': 'airflow',
+    'start_date': pendulum.today("UTC").add(days=-1),
+    'schedule': "@monthly",
+    'catchup': False,
+    'max_active_runs': 1,
+}
+
+dag = DAG(
+    'cleanup_logs',
+    default_args=default_args,
+    description='DAG to clean up old log directories',
+    schedule_interval='@monthly',
+)
+
+cleanup_command = f"""
+    logs_dir="{logs_dir}"
+    current_date=$(date "+%Y-%m-%dT%H:%M:%S.%N%z")
+
+    find "$logs_dir" -type f -path "*/run_id=*/task_id=*" -name "attempt=*.log" -exec sh -c '
+        run_id=$(dirname "$1" | grep -oP "run_id=\K[^/]+")
+        task_id=$(dirname "$1" | grep -oP "task_id=\K[^/]+")
+        if [ $(date -d "$run_id" "+%s") -lt $(date -d "$2" "+%s" --date="-30 days") ]; then
+            echo "Deleting: $1"
+            rm "$1"
+        fi
+    ' _ {{}} "$current_date" \;
+"""
+
+cleanup_task = BashOperator(
+    task_id='cleanup_logs',
+    bash_command=cleanup_command,
+    dag=dag,
+)
+
+cleanup_task
```

Comment:
Why don't we keep the same structure as the other dags? This is a different way to define it :)